branchpro/apps/_dash_app.py

```diff
@@ -150,6 +150,51 @@ def parse_contents(self, contents, filename):
         else:
             return html.Div(['Loaded data from: {}'.format(filename)]), df
 
+    def parse_interval_contents(self, contents, filename):
```

Comment:
I don't think it's ideal to duplicate all the code for string processing and json parsing in `parse_contents` and `parse_interval_contents`. One option would be to have one method with optional arguments, say `required_columns` and `required_dim`, and run the checks based on what the user provides. Alternatively, you could write a single _method to hold the filename splitting, base64 decoding, and pandas parsing, and two public methods to handle the particulars of each upload type. I'm happy with whichever makes more sense to you.